Gems/PhysX/Code/Editor/EditorJointConfiguration.cpp

```diff
@@ -411,13 +412,14 @@ namespace PhysX
             m_leadEntity);
         if (entity)
         {
+            const bool leadEntityHasRigidActor =
+                (entity->FindComponent<PhysX::EditorRigidBodyComponent>() ||
+                 entity->FindComponent<PhysX::EditorColliderComponent>() ||
+                 entity->FindComponent<PhysX::EditorShapeColliderComponent>());
+
             AZ_Warning("EditorJointComponent",
-                entity->FindComponent<PhysX::EditorRigidBodyComponent>() != nullptr,
-                "Please add a rigid body component to Entity %s. Joints do not work with a lead entity without a rigid body component.",
-                entity->GetName().c_str());
-            AZ_Warning("EditorJointComponent",
-                entity->FindComponent<PhysX::EditorColliderComponent>() != nullptr,
-                "Please add a collider component to Entity %s. Joints do not work with a lead entity without a collider component.",
+                leadEntityHasRigidActor,
+                "Please add a rigid body or collider component to Entity %s. Joints do not work with a lead entity without a rigid actor.",
```

Comment:
Is "rigid actor" the right term? maybe to not confuse it with skeletal actors this could be "rigid body"?

Reply:
Actor is the terminology which PhysX uses. That has always been kind of an annoying clash with the EMFX terminology.

Reply:
I see, quite annoying. Thanks.

Reply:
Having said that, this warning message might be quite confusing for users, since the fact that a collider component without a rigid body component creates a static rigid body at runtime is often misunderstood (and something we'd like to work on improving). Maybe the message could be rephrased to something like 

"Joints require either a dynamic or static rigid body on the lead entity. Please add either a rigid body component (to create a dynamic rigid body) or a collider component (to create a static rigid body) to entity %s"